asv_bench/benchmarks/benchmarks.py

```diff
@@ -422,3 +422,49 @@ def time_fillna(self, shape, limit, inplace):
             execute(self.df)
         else:
             execute(self.df.fillna(**kw))
+
+
+class BaseTimeValueCounts:
+    subset_params = {
+        "all": lambda shape: shape[1],
+        "half": lambda shape: shape[1] // 2,
+    }
+
+    def setup(self, shape, subset="all"):
+        try:
+            subset = self.subset_params.get(subset)
+        except KeyError:
+            raise KeyError(
+                f"Invalid value for 'subset={subset}'. Allowed: {list(self.subset_params.keys())}"
+            )
+        ncols = subset(shape)
+        self.df, _ = generate_dataframe(
+            ASV_USE_IMPL,
+            "int",
+            *shape,
+            RAND_LOW,
+            RAND_HIGH,
+            groupby_ncols=ncols,
+            count_groups=GROUPBY_NGROUPS[ASV_DATASET_SIZE],
+        )
+        self.subset = self.df.iloc[:, :ncols].columns.tolist()
```

Comment:
Maybe `self.subset = self.df.columns[:ncols]`? It's faster.

Reply:
applied